File: app/src/main/java/com/androidexamples/app/util/Animations.java
package com.androidexamples.app.util;

import android.content.Context;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.view.animation.ScaleAnimation;

/**
 * Created by marcos on 25/12/2016.
 */

public class Animations {

    public static void setSlideInLeftAnimation(View viewToAnimate, int currentPosition, int lastLoadedViewPosition, boolean alwaysAnimate, Context context) {
        if(alwaysAnimate || currentPosition > lastLoadedViewPosition) {
            Animation animation = AnimationUtils.loadAnimation(context, android.R.anim.slide_in_left);
            viewToAnimate.startAnimation(animation);
        }
    }

    public static void setSlideOutRightAnimation(View viewToAnimate, int currentPosition, int lastLoadedViewPosition, boolean alwaysAnimate, Context context) {
        if(alwaysAnimate || currentPosition > lastLoadedViewPosition) {
            Animation animation = AnimationUtils.loadAnimation(context, android.R.anim.slide_out_right);
            viewToAnimate.startAnimation(animation);
        }
    }

    public static void setFadeInAnimation(View viewToAnimate, int currentPosition, int lastLoadedViewPosition, boolean alwaysAnimate, Context context) {
        if(alwaysAnimate || currentPosition > lastLoadedViewPosition) {
            Animation animation = AnimationUtils.loadAnimation(context, android.R.anim.fade_in);
            animation.setDuration(Constants.FADE_DURATION);
            viewToAnimate.startAnimation(animation);
        }
    }

    public static void setFadeOutAnimation(View viewToAnimate, int currentPosition, int lastLoadedViewPosition, boolean alwaysAnimate, Context context) {
        if(alwaysAnimate || currentPosition > lastLoadedViewPosition) {
            Animation animation = AnimationUtils.loadAnimation(context, android.R.anim.fade_out);
            animation.setDuration(Constants.FADE_DURATION);
            viewToAnimate.startAnimation(animation);
        }
    }

    public static void setScaleAnimation(View viewToAnimate, int currentPosition, int lastLoadedViewPosition, boolean alwaysAnimate, Context context) {
        if(alwaysAnimate || currentPosition > lastLoadedViewPosition) {
            ScaleAnimation scaleAnimation = new ScaleAnimation(0.0f, 1.0f, 0.0f, 1.0f, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f);
            scaleAnimation.setDuration(Constants.SCALE_FADE_DURATION);
            viewToAnimate.startAnimation(scaleAnimation);
        }
    }
}

File: app/src/main/java/com/androidexamples/app/utils/Animations.kt
package com.androidexamples.app.utils

import android.content.Context
import android.view.View
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import android.view.animation.ScaleAnimation

/**
 * Created by marcos on 25/12/2016.
 */

object Animations {

    fun setSlideInLeftAnimation(viewToAnimate: View, currentPosition: Int, lastLoadedViewPosition: Int, alwaysAnimate: Boolean, context: Context) {
        if (alwaysAnimate || currentPosition > lastLoadedViewPosition) {
            val animation = AnimationUtils.loadAnimation(context, android.R.anim.slide_in_left)
            viewToAnimate.startAnimation(animation)
        }
    }

    fun setSlideOutRightAnimation(viewToAnimate: View, currentPosition: Int, lastLoadedViewPosition: Int, alwaysAnimate: Boolean, context: Context) {
        if (alwaysAnimate || currentPosition > lastLoadedViewPosition) {
            val animation = AnimationUtils.loadAnimation(context, android.R.anim.slide_out_right)
            viewToAnimate.startAnimation(animation)
        }
    }

    fun setFadeInAnimation(viewToAnimate: View, currentPosition: Int, lastLoadedViewPosition: Int, alwaysAnimate: Boolean, context: Context) {
        if (alwaysAnimate || currentPosition > lastLoadedViewPosition) {
            val animation = AnimationUtils.loadAnimation(context, android.R.anim.fade_in)
            animation.duration = Constants.FADE_DURATION.toLong()
            viewToAnimate.startAnimation(animation)
        }
    }

    fun setFadeOutAnimation(viewToAnimate: View, currentPosition: Int, lastLoadedViewPosition: Int, alwaysAnimate: Boolean, context: Context) {
        if (alwaysAnimate || currentPosition > lastLoadedViewPosition) {
            val animation = AnimationUtils.loadAnimation(context, android.R.anim.fade_out)
            animation.duration = Constants.FADE_DURATION.toLong()
            viewToAnimate.startAnimation(animation)
        }
    }

    fun setScaleAnimation(viewToAnimate: View, currentPosition: Int, lastLoadedViewPosition: Int, alwaysAnimate: Boolean, context: Context) {
        if (alwaysAnimate || currentPosition > lastLoadedViewPosition) {
            val scaleAnimation = ScaleAnimation(0.0f, 1.0f, 0.0f, 1.0f, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f)
            scaleAnimation.duration = Constants.SCALE_FADE_DURATION.toLong()
            viewToAnimate.startAnimation(scaleAnimation)
        }
    }
}
